oauthproxy_test.go

```diff
@@ -1776,10 +1780,117 @@ func TestGetRedirect(t *testing.T) {
 			url:              proxy.ProxyPrefix + "/foo/bar",
 			expectedRedirect: "/",
 		},
+		{
+			name: "proxied request outside of ProxyPrefix redirects to proxied URL",
+			url:  "https://oauth.example.com/foo/bar",
+			headers: map[string]string{
+				"X-Forwarded-Proto": "https",
+				"X-Forwarded-Host":  "a-service.example.com",
+				"X-Forwarded-Uri":   "/foo/bar",
+			},
+			expectedRedirect: "https://a-service.example.com/foo/bar",
+		},
+		{
+			name: "proxied request with query preserves query",
+			url:  "https://oauth.example.com/foo?bar",
+			headers: map[string]string{
```

Comment:
Let's leave these top 3 tests with `nil` headers so we can preserve their original testing purpose for the majority of users who won't have these headers.

Reply:
Alright.